include/logger/loggerConfig.h

```diff
@@ -272,59 +276,104 @@ typedef struct _PWMConfig {
 #define DEFAULT_PWM_CONFIG {DEFAULT_PWM_CHANNEL_CONFIG, MODE_PWM_FREQUENCY, MODE_LOGGING_PWM_DUTY, DEFAULT_PWM_DUTY_CYCLE, DEFAULT_PWM_PERIOD}
 
 
-#define OBD2_CHANNELS 20
+#define DEFAULT_CAN_BAUD_RATE 500000
+
+typedef struct _CANConfig {
+    unsigned char enabled;
+    int baud[CONFIG_CAN_CHANNELS];
+#if CAN_SW_TERMINATION == true
+    bool termination[CONFIG_CAN_CHANNELS];
+#endif
+} CANConfig;
+
+/* define max offsets and length for CAN mappings */
+#define MAX_CAN_MAPPING_OFFSET_BYTES 7
+#define MAX_CAN_MAPPING_LENGTH_BYTES 4
+
+enum CANMappingEndian {
+    CANMappingEndian_Big = 0,
+    CANMappingEndian_Little
+};
+
+
+typedef struct _CANMapping {
+    /* The standard channel configuration */
+    ChannelConfig channel_cfg;
+
+    /* CAN ID to match this mapping */
+    uint32_t can_id;
+
+    /* mask to apply to ID */
+    uint32_t can_mask;
+
+    /* multiplier for conversion formula */
+    float multiplier;
+
+    /* divider for conversion formula */
+    float divider;
```

Comment:
I agree in principal and that's how I had it originally.  However, it's apparently a thing in the CAN mapping world where explicit dividers are specified and CAN .dbc files have an explicit divider stated, even though it could be represented as a fractional multiplier.  So, I added it to be compatible with the quasi-standard.